node/src/components/consensus/protocols/zug.rs

```diff
@@ -2416,6 +2444,14 @@ where
     }
 }
 
+fn outcomes_or_disconnect<C: Context>(
+    result: Result<ProtocolOutcomes<C>, FaultySender>,
+) -> ProtocolOutcomes<C> {
+    result.unwrap_or_else(|sender| vec![ProtocolOutcome::Disconnect(sender.0)])
+}
+
+struct FaultySender(NodeId);
```

Comment:
Nit: I'd move it somewhere nearer the top of the file.

Reply:
I'll do that in another PR, then, I won't create a commit only for moving a line.